weldx/core.py

```diff
@@ -458,21 +461,21 @@ def _interp_time_discrete(self, time: Time) -> xr.DataArray:
 
     def _interp_time_expression(self, time: Time, time_unit: str) -> xr.DataArray:
         """Interpolate the time series if its data is a mathematical expression."""
-        time_q = time.as_quantity(unit=time_unit)
+        time_q = xr.DataArray(time.as_quantity(unit=time_unit), dims=["time"])
 
-        if len(self.shape) > 1 and np.iterable(time_q):
-            while len(time_q.shape) < len(self.shape):
-                time_q = time_q[:, np.newaxis]
+        # if len(self.shape) > 1 and np.iterable(time_q):
+        #    while len(time_q.shape) < len(self.shape):
+        #        time_q = time_q[:, np.newaxis]
 
         # evaluate expression
         data = self._data.evaluate(**{self._time_var_name: time_q})
-        data = data.astype(float).to_reduced_units()  # float conversion before reduce!
+        # data = data.astype(float).to_reduced_units()  # float conversion before reduce!
 
         # create data array
-        if not np.iterable(data):  # make sure quantity is not scalar value
-            data = np.expand_dims(data, 0)
-
-        return self._create_data_array(data, time)
+        # if not np.iterable(data):  # make sure quantity is not scalar value
+        #    data = np.expand_dims(data, 0)
+        data = data.assign_coords({"time": time.as_timedelta_index()})
```

Comment:
do we loose the reference_timestamp information here?

I think
```suggestion
        data = data.assign_coords({"time": time.as_data_array()})
```
should also work (or manually setting the reference time)

Reply:
Should work. However, there is this `()` shape problem again I have to fix. Apart from that I think we are good to go

Reply:
Updated it. Just needed to adjust the implementation of `Time.as_data_array` because single time values caused failures. I replaced the usage of `self._time` with `self.as_pandas_index` when constructing the `DataArray`